scripts/build/builders/telink.py

```diff
@@ -60,35 +60,31 @@ def __init__(self,
         self.board = board
 
     def generate(self):
+        if os.path.exists(self.output_dir):
+            return
 
-        if not os.path.exists(self.output_dir):
-            cmd = 'export ZEPHYR_BASE="$TELINK_ZEPHYR_BASE"\n'
+        if not self._runner.dry_run:
+            # Zephyr base
+            if 'TELINK_ZEPHYR_BASE' not in os.environ:
+                raise Exception("Telink builds require TELINK_ZEPHYR_BASE")
 
-            if not self._runner.dry_run:
+        cmd = 'export ZEPHYR_BASE="$TELINK_ZEPHYR_BASE"\n'
 
-                # Zephyr base
-                if 'TELINK_ZEPHYR_BASE' not in os.environ:
-                    # TODO: remove once variable in all images
-                    cmd = ''
-                    if 'ZEPHYR_BASE' not in os.environ:
-                        raise Exception(
-                            "Telink builds require TELINK_ZEPHYR_BASE or ZEPHYR_BASE to be set")
+        if 'TELINK_ZEPHYR_SDK_DIR' in os.environ:
+            cmd += 'export ZEPHYR_SDK_INSTALL_DIR="$TELINK_ZEPHYR_SDK_DIR"\n'
 
-            # TODO: TELINK_ZEPHYR_SDK_DIR should be used for compilation and
-            # NOT hardcoding of zephyr-sdk-0.13.0
-            cmd += '''
+        cmd += '''
 export ZEPHYR_TOOLCHAIN_VARIANT=zephyr
-export ZEPHYR_SDK_INSTALL_DIR="$ZEPHYR_BASE/../../zephyr-sdk-0.13.0"
 source "$ZEPHYR_BASE/zephyr-env.sh";
 west build --cmake-only -d {outdir} -b {board} {sourcedir}
         '''.format(
-                outdir=shlex.quote(
-                    self.output_dir), board=self.board.GnArgName(), sourcedir=shlex.quote(
-                    os.path.join(
-                        self.root, 'examples', self.app.ExampleName(), 'telink'))).strip()
+            outdir=shlex.quote(
+                self.output_dir), board=self.board.GnArgName(), sourcedir=shlex.quote(
+                os.path.join(
+                    self.root, 'examples', self.app.ExampleName(), 'telink'))).strip()
 
-            self._Execute(['bash', '-c', cmd],
-                          title='Generating ' + self.identifier)
+        self._Execute(['bash', '-c', cmd],
```

Comment:
Why does this use 'bash' which isn't portable to all platforms ?

Reply:
We assume 'we compile on linux or mac'.
We could update to just 'sh' eventually, however I am not convinced our commands will work. Unfortunately quitte a few need environment variables in the build systtem since they use cmake/zephyr insted of gn arguments.

Reply:
'sh' is also a Unix thing.

We should write the python scripts to be Windows compatible where there is no Unix shell. That's one of the reasons _why_ we prefer python tooling.

Reply:
Agree, but probably for future PRs. The reason this does the -c instead of the python exec (which it could and then environment variables could be passed in) is that the script is trying to have a `--dry-run` execution that outputs "run these commands on the command line and you will get a build". For this reason I am doing 'cp' instead of 'shutil.FileCopy' and I do 'bash' instead of 'suprocess.Popen' or similar.

Figuring out and accepting full-python driven execution should be tackled separately (I am unsure if we will in the near/medium future).